lib/packwerk/cache_deprecated_references.rb

```diff
@@ -0,0 +1,45 @@
+# typed: true
```

Comment:
I think it would be good that new files to this codebase are in `strict` mode. A lot of the benefits from sorbet only arise if we have enough type information available.

Reply:
While that would be nice, these methods are extracted from what used to be UpdatingDeprecatedReferences class and unfortunately I don't have enough context to know what the signatures for these classes would be.

Reply:
I think it is important that we continue adopting sorbet in this codebase because it will improve the maintainability of the code over time. It is a bit inconvenient to look up the signatures but it is beneficial in the long term.

You can add type signatures to this class pretty easily and change to strict:

```ruby
   
    sig { params(root_path: String, deprecated_references: T::Hash[String, Packwerk::DeprecatedReferences]) }
    def initialize(root_path, deprecated_references = {})
      @root_path = root_path
      @deprecated_references = T.let(deprecated_references, T::Hash[String, Packwerk::DeprecatedReferences])
    end

    sig { params(package: Package).returns(Packwerk::DeprecatedReferences) } 
    def deprecated_references_for(package)
      @deprecated_references[package] ||= Packwerk::DeprecatedReferences.new(
        package,
        deprecated_references_file_for(package),
      )
    end

    sig { params(package: Package).returns(String) } 
    def deprecated_references_file_for(package)
      File.join(@root_path, package.name, "deprecated_references.yml")
    end
```

Reply:
I attempted to do this as it seemed like it would be low hanging fruit we could ship with this PR, but these signatures generated some really weird typecheck errors that should be further investigated in another PR. For instance, in `deprecated_references_for` package param seems to be both a String and Package. It's a little weird and I really think necessitates further investigation.